### ISSUE TITLE
Call to FSM Backend PieceManager SealPieceIntoNewSector never returns

### ISSUE THREAD

**Issue by hannahhoward** [filecoin-project/go-filecoin#4112]
**Describe the bug**:

When creating a deal on a local devnet (I used a miner and a client node, using scripts here https://gist.github.com/hannahhoward/ea1f9f5eabe420dcfaae1aa7d7bf5ea9 -- I was using the `feat/deals-integration-work` branch cause it has a `deals list` command but master should also work) -- it miner never completes the call to piecemanager.SealPieceIntoNewSector.

I've attached a log which shows relevant messages:
![Screen Shot 2020-05-13 at 1 12 58 PM](https://user-images.githubusercontent.com/1450680/81861237-c8e07e80-951c-11ea-8bdd-b737829cb77c.png)

Based on the log messages in storage-fsm.Sealing.SealPiece,
we should see a message that reads: 	
"Start sealing ~sector-id here"
before we start actually sealing.

But we never do even after 10-15 minutes.

Seems like it stuck in m.sealer.AddPiece

**Comment by hannahhoward** [comment]
FWIW, I see that on the lotus side theres a call to Sealing.AllocatePiece before the call to SealPiece

**Comment by hannahhoward** [comment]
Follow-up, running with debug information for "advmgr":

```
2020-05-13T14:11:24.627-0700	INFO	sectors	storage-fsm@v0.0.0-20200508212339-4980cb4c92b1/sealing.go:108	Seal piece for deal 2
2020-05-13T14:11:24.627-0700	DEBUG	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:373	sched: not scheduling on worker 0; not enough physical memory - need: 18438M, have 16384M
```

**Comment by anorth** [comment]
Regarding memory, are you sure the node is configured with 2k sectors? The default sector sizes very recently changed to be imported from specs-actors, and there is a [config setting in go-filecoin](https://github.com/filecoin-project/go-filecoin/blob/fdb4159d704c7f44f1dba9ebf257eddd0151746c/internal/pkg/config/config.go#L250) to override those defaults. 2k is not a default.

There's is no CLI for this setting at present.

**Comment by hannahhoward** [comment]
dditional debugging into the sector storage code has revealed that apparently, the root is: I had too many programs open? that said, it did not get us to success unfortunately. Now there’s an error on pre-commit.

One of the numbers that’s factor into whether it will run a job is apprently res.MemReserved — AKA memory already allocated. so memReserved + used on existing jobs + needed for this job. It was already getting the right number for prooftype (2K) but apparently my res.MemReserved was above my physical memory. (probably due to chatting on her and stuff)

But, to fail silently due to out of physical memory in a way that completely blocks the process (reminder the warning only shows with debug logging), that seems like it should be addressed.

Will file issue on sector storage, have no idea what is going on with HandlePreCommit

```
2020-05-13T17:53:14.611-0700	WARN	storage connector	storage_market/provider.go:138	finished a seal call for PieceCID: bafk4chzawfxqh64jktkfjt6l4zlqkewrgkwvu6rpb4n5ht3hhhlz73fefqrq
2020-05-13T17:53:14.612-0700	INFO	sectors	storage-fsm@v0.0.0-20200508212339-4980cb4c92b1/states.go:19	performing filling up rest of the sector...	{"sector": "6"}
2020-05-13T17:53:14.612-0700	WARN	sectors	storage-fsm@v0.0.0-20200508212339-4980cb4c92b1/states.go:38	Creating 2 filler pieces for sector 6
2020-05-13T17:53:14.612-0700	INFO	sectors	storage-fsm@v0.0.0-20200508212339-4980cb4c92b1/garbage.go:24	Pledge {106 6}, contains [508]
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:225	{MinMemory:2048 MaxMemory:2048 Threads:1 CanGPU:false BaseMinMemory:2048}
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:370	{MinMemory:2048 MaxMemory:2048 Threads:1 CanGPU:false BaseMinMemory:2048}
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:373	res.MemReserved 10604089344 + active.memUsedMin 0 + needRes.MinMemory 2048 + needRes.BaseMinMemory 2048
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:370	{MinMemory:2048 MaxMemory:2048 Threads:1 CanGPU:false BaseMinMemory:2048}
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:373	res.MemReserved 10604089344 + active.memUsedMin 0 + needRes.MinMemory 2048 + needRes.BaseMinMemory 2048
2020-05-13T17:53:14.612-0700	DEBUG	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/localworker.go:66	acquired sector {106 6} (e:1; a:0): {{0 0} /Users/hannah/.filecoin_sectors/unsealed/s-t0106-6  }
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:225	{MinMemory:2048 MaxMemory:2048 Threads:1 CanGPU:false BaseMinMemory:2048}
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:370	{MinMemory:2048 MaxMemory:2048 Threads:1 CanGPU:false BaseMinMemory:2048}
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:373	res.MemReserved 10604089344 + active.memUsedMin 0 + needRes.MinMemory 2048 + needRes.BaseMinMemory 2048
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:370	{MinMemory:2048 MaxMemory:2048 Threads:1 CanGPU:false BaseMinMemory:2048}
2020-05-13T17:53:14.612-0700	WARN	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/sched.go:373	res.MemReserved 10604089344 + active.memUsedMin 0 + needRes.MinMemory 2048 + needRes.BaseMinMemory 2048
2020-05-13T17:53:14.612-0700	DEBUG	advmgr	sector-storage@v0.0.0-20200508203401-a74812ba12f3/localworker.go:66	acquired sector {106 6} (e:1; a:0): {{0 0} /Users/hannah/.filecoin_sectors/unsealed/s-t0106-6  }
2020-05-13T17:53:14.613-0700	ERROR	sectors	storage-fsm@v0.0.0-20200508212339-4980cb4c92b1/states.go:59	handlePreCommit1: api error, not proceeding: getting deal 2 for piece 0: deal 2 not found
2020-05-13T17:53:37.004-0700	INFO	mining	mining/worker.go:183	Worker.Mine
```

**Comment by anorth** [comment]
Will be resolved by https://github.com/filecoin-project/go-filecoin/pull/4119

**Comment by hannahhoward** [comment]
@anorth not a go filecoin issue, but I want to clarify there is definitely a seperate bug in sector-storage where if you fail the memory check, AddPiece will hang in definitely. I'm not sure if the memory check is itself an issue, but the scheduler definitely has an issue with not returning anything. I will file with them.

**Comment by hannahhoward** [comment]
Closing in favor of: https://github.com/filecoin-project/sector-storage/issues/35

Note that SealPieceIntoSector actually returns as long as the memory check doesn't fail, so #4119 is solving a subsequent issue.
